core/domain/rights_manager.py

```diff
@@ -1068,7 +1081,7 @@ def assign_role_for_exploration(
     """
     _assign_role(
         committer, assignee_id, new_role, exploration_id,
-        constants.ACTIVITY_TYPE_EXPLORATION)
+        constants.ACTIVITY_TYPE_EXPLORATION, allow_assigning_any_role=True)
```

Comment:
Just to check: is the intention long-term to always allow reassigning now?

Reply:
(Reply to this thread as I have written the design doc)
You mean: whether this is going to be a long-term feature and not a one-off feature so that we can remove some users from some exploration, right? (Sorry if I misunderstood!) 

@BenHenning, Yeah as per the requirements, we want to have this feature so that owner can add/remove roles based on the collaborator's activity and availability.

Reply:
Got it--that is what I was asking. Thanks @DubeySandeep!